Tests failing to build in trust store mock

Builds are failing in `xenial` branch with the following error:

```
[ 77%] Building CXX object tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/tst_trust_store_model.cpp.o

cd /build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests/plugins/security-privacy && /usr/lib/ccache/x86_64-linux-gnu-g++   -DDATA_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/data\" -DI18N_DOMAIN=\"ubuntu-system-settings\" -DMANIFEST_DIR=\"data\" -DPLUGIN_MANIFEST_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/data\" -DPLUGIN_MODULE_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests\" -DPLUGIN_PRIVATE_MODULE_DIR=\"/usr/lib/x86_64-linux-gnu/ubuntu-system-settings/private\" -DPLUGIN_QML_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests\" -DQML_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests\" -DQML_TEST_DIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests\" -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_QML_LIB -DQT_TESTCASE_BUILDDIR=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu\" -DQT_TESTLIB_LIB -DSYSTEM_IMAGE_DBUS_TEMPLATE=\"/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/autopilot/ubuntu_system_settings/tests/systemimage.py\" -DTESTS -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/lib -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/src -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/../src -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libqtdbusmock-1 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/include/x86_64-linux-gnu/qt5 -I/usr/include/libqtdbustest-1 -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu/tests/plugins/security-privacy -I/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/../../../plugins/security-privacy -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtDBus -isystem /usr/include/x86_64-linux-gnu/qt5/QtQml -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtTest  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -std=c++0x -fno-permissive -pedantic -Wall -Wextra   -fPIC -std=gnu++11 -o CMakeFiles/tst-trust-store-model.dir/tst_trust_store_model.cpp.o -c /build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/tst_trust_store_model.cpp

/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/tst_trust_store_model.cpp: In member function 'void TrustStoreModelTest::init()':

/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/tst_trust_store_model.cpp:242:54: error: invalid new-expression of abstract class type '{anonymous}::mock::Store'

     m_store = std::shared_ptr<mock::Store>(new mock::Store);

                                                      ^

/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/tst_trust_store_model.cpp:33:8: note:   because the following virtual functions are pure within '{anonymous}::mock::Store':

 struct Store: public core::trust::Store

        ^

In file included from /build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/tests/plugins/security-privacy/tst_trust_store_model.cpp:24:0:

/usr/include/core/trust/store.h:181:18: note: 	virtual void core::trust::Store::remove_application(const string&)

     virtual void remove_application(const std::string& id) = 0;

                  ^

tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/build.make:65: recipe for target 'tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/tst_trust_store_model.cpp.o' failed

make[4]: *** [tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/tst_trust_store_model.cpp.o] Error 1

make[4]: Leaving directory '/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu'

CMakeFiles/Makefile2:7870: recipe for target 'tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/all' failed

make[3]: *** [tests/plugins/security-privacy/CMakeFiles/tst-trust-store-model.dir/all] Error 2

make[3]: Leaving directory '/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu'

Makefile:141: recipe for target 'all' failed

make[2]: *** [all] Error 2

make[2]: Leaving directory '/build/varlibjenkinsworkspaceubportssystemsettingsPRubuntusystemsettingsubportsgbpcdbdsc/obj-x86_64-linux-gnu'

dh_auto_build: make -j1 returned exit code 2
```

--- a/tests/plugins/security-privacy/tst_trust_store_model.cpp
+++ b/tests/plugins/security-privacy/tst_trust_store_model.cpp
@@ -176,6 +176,13 @@ struct Store: public core::trust::Store
     {
     }
 
+    void remove_application(const std::string& appid)
+    {
+        m_allRequests.remove_if([&appid](core::trust::Request &r) {
+                return r.from == appid;
+            });
+    }
+
     std::shared_ptr<core::trust::Store::Query> query()
     {
         auto query =

